prime-router/docs/proposals/0017-metadata-tables-cleanup.md

```diff
@@ -0,0 +1,441 @@
+3. Create a script that will migrate existing legacy data from the `covid_result_metadata` and `elr_result_metadata` tables into the new table structure
+4. Create two materialized views, one named `covid_result_metadata` and one named `elr_result_metadata` that will exactly replicate the layout of the current tables in order to prevent breakage of any dashboards in Metabase or existing queries
+    - As part of this, the two existing tables will be renamed but preserved for some period of time until we can be sure we have not lost any data
+
+#### Phase 2
+This work will happen as a second step, hopefully in Q1 2023, and reflect a move towards more independent microservices. On ingestion, ReportStream will invoke a separate Azure function that will queue up messages to process, which will then be saved into our metadata tables. The goal is to separate the concerns of processing from the goal of tracking metadata from the messages we receive.
+
+1. Create a new independent queue in distinct namespace that will hold the raw messages to be processed
+2. Create a separate Azure function that will accept the message and process the contents of the queue
+
+### Discussion
+By implementing these changes, repeated information like facility details, provider information, as well as geographic information can be cleaned up and more easily indexed. In addition, we are able to collect more complex information in this structure, for example a message having multiple AOE questions, or multiple specimens, or multiple ordered tests. We also can begin to build in more advanced data verification, such as checking CLIAs or NPIs. We also have the option to extend out features like our address enrichment by adding things like census tract details and ZIP+4.  
+
+### Link to the proposed changes
+You can review the currently proposed SQL changes here, or at the bottom in Appendix 1:
+
+
+## Further Work
+Once feedback has been provided about the structure of the tables and the plan above, any proposed changes will go through an acceptance/rejection process, and then the actual changes will be put into a PR for a final review. This document has been created on 3 November 2022, and will be reviewed at the developer sync on Monday, 7 November 2022 with a targeted proposal acceptance date of 14 November 2022. At that point the work will begin on Phase 1 with an expected completion date of 21 November 2022. Start and completion date for Phase 2 TBD.
+
+It should also be noted that this work is not to make a system of record for the CDC or for our STLT partners. We are still only acting as a hub for the transferring of data. That said, having the ability to provide usage and demographic data to our senders and receivers means we provide a richer and better experience for our users, and it empowers our support staff.
+
+## Appendix 1
+
+```postgresql
```

Comment:
Is there a definitive set of requirement we are trying to meet here? I know the covid metadata table is used and gives guidance on that, but the elr metadata table is not used and may not have all the data / may have extra data